packages/camera/camera_avfoundation/example/ios/RunnerTests/FLTCamPhotoCaptureTests.m

```diff
@@ -110,10 +110,10 @@ - (void)testCaptureToFile_mustReportFileExtensionWithHeifWhenHEVCIsAvailableAndF
   id mockSettings = OCMClassMock([AVCapturePhotoSettings class]);
   OCMStub([mockSettings photoSettingsWithFormat:OCMOCK_ANY]).andReturn(settings);
 
-  NSString *filePath = @"test";
+  NSString *filePath = @"test.heif";
   id mockResult = OCMClassMock([FLTThreadSafeFlutterResult class]);
   OCMStub([mockResult sendSuccessWithData:filePath]).andDo(^(NSInvocation *invocation) {
-    XCTAssertEqualObjects(cam.fileExtension, @"heif");
+    XCTAssertEqualObjects([filePath pathExtension], @"heif");
```

Comment:
This is odd - filePath is a local variable hard coded to be "test.heif". 

Can you do something like: 

```
OCMStub([mockResult sendSuccessWithData: OCARG_ANY]).andDo(^(NSInvocation *invocation)) {
  NSString *pathArgument;
  [invocation getArgument: &pathArgument at index...];
  XCTAssert( pathArgument contains heif )
});
```

Reply:
I'm not sure if this will work. Because here I will need to pass the path to  delegate.completionHandler;

Reply:
I changed this part for testing and it worked as expected:

```
  FLTSavePhotoDelegate *savePhotoDelegate = [[FLTSavePhotoDelegate alloc]
           initWithPath:path
                ioQueue:self.photoIOQueue
      //changed here to use path directly only to test,because completionHandler needs to path: 
       completionHandler:^(NSString *_Nullable path2, NSError *_Nullable error) {
        typeof(self) strongSelf = weakSelf;
        if (!strongSelf) return;
        dispatch_async(strongSelf.captureSessionQueue, ^{
          // cannot use the outter `strongSelf`
          typeof(self) strongSelf = weakSelf;
          if (!strongSelf) return;
          [strongSelf.inProgressSavePhotoDelegates removeObjectForKey:@(settings.uniqueID)];
        });

        if (error) {
          [result sendError:error];
        } else {
          NSAssert(path, @"Path must not be nil if no error.");
          [result sendSuccessWithData:path];
        }
      }];
```
Maybe I should change the FLTSavePhotoDelegate or is there some way I'm not seeing?